lib/documents/query/query_builder.rb

```diff
@@ -0,0 +1,49 @@
+require 'utilities/linked_list'
+require 'database/exceptions'
+require 'utilities/observable'
+require 'constants/documents'
+
+module RavenDB
+  class QueryBuilder
+    include Observable
+
+    def initialize(index_name = nil, collection_name = nil, id_property_name = nil)
+      unless index_name.nil? && collection_name.nil?
+        from(index_name, collection_name)
+      end
+
+      @group_by_tokens = LinkedList.new
```

Comment:
You can use the standard list on Ruby, no real requirement to make this linked list.